clang-tools-extra/test/clang-read-diagnostics/error_and_warning.c

```diff
@@ -0,0 +1,15 @@
+// REQUIRES: clang
+// RUN: not %clang %s -serialize-diagnostics %s.diag
```

Comment:
recommend putting the new file under `%t` instead of `%s`, so this doesn't pollute the source dir.